internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlyStoreConfigRepository.java

```diff
@@ -114,12 +99,26 @@ public Optional<StoreConfig> getStoreConfig(String storeName) {
     if (systemStoreType != null && systemStoreType.equals(VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE)) {
       veniceStoreName = VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.extractRegularStoreName(storeName);
     }
-    StoreConfig config = storeConfigMap.get().get(veniceStoreName);
-    if (config != null) {
-      return Optional.of(config.cloneStoreConfig());
-    } else {
-      return Optional.empty();
+
+    if (availableStoreSet.get().contains(veniceStoreName)) {
+      StoreConfig config = loadedStoreConfigMap.get(veniceStoreName);
+      if (config == null) {
+        // lazy fetch from ZK and attach watch
+        config = accessor.getStoreConfig(veniceStoreName);
+        if (config != null) {
```

Comment:
Minor improvement:
Shall we log a warn message if the store is present in the store list, but the corresponding `storeConfig` is `null`?
normally, it should not happen.

Reply:
I think it's a great idea